pydicom/values.py

```diff
@@ -379,20 +379,18 @@ def convert_value(VR, raw_data_element, encoding=default_encoding):
         logger.debug('unable to translate tag %s with VR %s'
                      % (raw_data_element.tag, VR))
 
-        for convert_vr in convert_retry_VR_order:
-            vr = convert_vr
-            converter = converters[vr]
+        for vr in convert_retry_VR_order:
             if vr == VR:
                 continue
             try:
                 value = convert_value(vr, raw_data_element, encoding)
-                break
-            except Exception:
-                pass
-            else:
```

Comment:
On the second glance, I think I got the logic right - just the logging is missing for the fallback case. I just used direct return instead of the else branch, but the else branch is cleaner, I think.